garden/src/features/entrypoints/components/modals/PaperModal.tsx

```diff
@@ -46,11 +52,106 @@ const PaperModal = ({ edit, onSave, initialData, trigger }: PaperModalProps) =>
         })) || [],
       doi: initialData?.doi || "",
       citation: initialData?.citation || "",
+      url: initialData?.url || "",
     },
   });
 
+  const url = form.watch("url");
+
+  // Reset form when modal is opened or closed
+  useEffect(() => {
+    if (isOpen) {
+      // If we're editing, load the initial data
+      if (edit && initialData) {
+        form.reset({
+          title: initialData.title || "",
+          authors:
+            initialData.authors?.map((author) => ({
+              value: author,
+              label: author,
+            })) || [],
+          doi: initialData.doi || "",
+          citation: initialData.citation || "",
+          url: initialData.url || "",
+        });
+      } else {
+        // If we're adding a new paper, reset to empty values
+        form.reset({
+          title: "",
+          authors: [],
+          doi: "",
+          citation: "",
+          url: "",
+        });
+      }
+      previousUrlRef.current = "";
```

Comment:
I'm a little surprised to see a useref in here. Maybe it's justified, but it's surprising

Reply:
turns out it wasn't justified at all